frontend/src/components/admin/AdminDashboard.js

```diff
@@ -1,18 +1,49 @@
-import React from "react";
+import React, { Component } from "react";
+import { Input, Button } from "antd";
 import { AuthorizationGuard } from "@/HOC/AuthorizationGuard";
 import * as Permission from "@/constants/permissions";
+import { downloadMineManagerHistory } from "@/actionCreators/partiesActionCreator";
 
 /**
  * @class AdminDashboard houses everything related to admin tasks, this is a permission-based route.
  */
 
-export const AdminDashboard = () => (
-  <div className="landing-page">
-    <div className="landing-page__header" />
-    <div className="landing-page__content">
-      <h1>Admin View</h1>
-    </div>
-  </div>
-);
+export class AdminDashboard extends Component {
+  constructor(props) {
+    super(props);
+    this.state = {
+      mineNo: "",
+    };
+  }
+
+  handleChange = (e) => {
+    this.setState({
+      mineNo: e.target.value,
+    });
+  };
+
+  handleDownload = () => {
+    downloadMineManagerHistory(this.state.mineNo, { window, document });
+  };
+
+  render() {
+    return (
+      <div className="landing-page">
+        <div className="landing-page__header" />
+        <div className="landing-page__content">
+          <h1>Admin View</h1>
+          <h2>Pull Mine Manager History</h2>
+          <Input
```

Comment:
Would this be better as a redux form and then we wouldn't have to worry about setting the state at all with mineNo?

Reply:
I knew you'd ask or have some comment about it! But no, I personally don't see any reason for this component to have application-wide state. The only state is the input field, which is relevant to this component only, does not need to persists between routes, etc. I think leaving out redux here makes sense (at this point, anyway).

We may choose to connect this to the store if/when we add more to the Admin dashboard.